src/omnisharp/server.ts

```diff
@@ -380,20 +381,32 @@ export class OmniSharpServer {
                 });
             }
 
-            // If there's more than one launch target, we start the server if one of the targets
-            // matches the preferred path. Otherwise, we fire the "MultipleLaunchTargets" event,
-            // which is handled in status.ts to display the launch target selector.
-            if (launchTargets.length > 1 && preferredPath) {
+            const defaultLaunchSolution = launchTargets.filter((a) => (a.isDefault === true));
+            // If there's more than one target, launch the server with the launch target that
+            // matches the default solution (if it's configured).
+            // The user can still manually switch to another solution from  the 'Omnisharp: Select Project command'
```

Comment:
Will it still remember the last selected project on the next start up (i.e. the `preferredPath`). Or, does setting this option effectively override it? I would expect `defaultLaunchSolution` to be used in the case where there isn't a `preferredPath`.

Reply:
If`omnisharp.defaultLaunchSolution` is set (to a valid existing solution), it will be used at start up. That's the intended behaviour. If it's not set then the last selected should be remembered and used at start up per the original logic.